paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java

```diff
@@ -96,13 +102,29 @@ private boolean isAutoTag(String tag) {
     }
 
     public boolean forceCreatingSnapshot() {
+        if (timeExtractor instanceof WatermarkExtractor && idlenessTimeout != null) {
+            return checkIdleTimeout();
+        }
         return timeExtractor instanceof ProcessTimeExtractor
                 && (nextTag == null
                         || isAfterOrEqual(
                                 LocalDateTime.now().minus(delay),
                                 periodHandler.nextTagTime(nextTag)));
     }
 
+    private boolean checkIdleTimeout() {
```

Comment:
inline this method into `forceCreatingSnapshot`, it is hard to understand from this method name `checkIdleTimeout `

Reply:
done.